packages/odyssey-react-mui/src/labs/NavRail/NavRail.tsx

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import styled from "@emotion/styled";
+import { memo } from "react";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../../OdysseyDesignTokensContext";
+import { OktaAura } from "./OktaAura";
+import { NavRailApp, NavRailAppProps } from "./NavRailApp";
+
+const NAV_RAIL_WIDTH = "64px";
+
+export type NavRailProps = {
```

Comment:
Add `visibilityState = invisible | loading | loaded`

### How should we handle the loading state?

Skeleton
- Enduser: Skeleton shows, then disappears (pop in during load)
- Admin: Skeleton shows, then gets populated

No Skeleton ✅ 
- Enduser: Skeleton does not show, then disappears
- Admin: Skeleton does not show, then nav rail pops in
- Could we mitigate this via storing in local storage to prevent pop in? Would pop in the first load, but would not pop in after that.
- Also consider that this is just for the MVP where it is only shown for the admin. In the future, when it is shown to all users, we could switch to skeleton.

Full screen loading screen (prevents pop in)
- Would block showing the app until we get the data. Unacceptable to downstream teams.

Keep side nav the same width regardless of app switcher
- No, would break design assumptions